rust/src/shards/gui/layouts/table.rs

```diff
@@ -0,0 +1,429 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright © 2022 Fragcolor Pte. Ltd. */
+
+use super::Table;
+use crate::shard::Shard;
+use crate::shards::gui::util;
+use crate::shards::gui::EguiId;
+use crate::shards::gui::BOOL_VAR_OR_NONE_SLICE;
+use crate::shards::gui::INT_VAR_OR_NONE_SLICE;
+use crate::shards::gui::PARENTS_UI_NAME;
+use crate::shardsc::SHType_Int;
+use crate::shardsc::SHType_Seq;
+use crate::shardsc::SHType_ShardRef;
+use crate::shardsc::SHType_String;
+use crate::types::common_type;
+use crate::types::Context;
+use crate::types::ExposedInfo;
+use crate::types::ExposedTypes;
+use crate::types::InstanceData;
+use crate::types::OptionalString;
+use crate::types::ParamVar;
+use crate::types::Parameters;
+use crate::types::Seq;
+use crate::types::ShardsVar;
+use crate::types::Type;
+use crate::types::Types;
+use crate::types::Var;
+use crate::types::ANYS_TYPES;
+use crate::types::SEQ_OF_SHARDS_TYPES;
+use std::cmp::Ordering;
+use std::ffi::CStr;
+
+lazy_static! {
+  static ref ANY_TABLE_TYPES: Vec<Type> = vec![common_type::any_table];
+  static ref SEQ_OF_ANY_TABLE: Type = Type::seq(&ANY_TABLE_TYPES);
+  static ref SEQ_OF_ANY_TABLE_TYPES: Vec<Type> = vec![*SEQ_OF_ANY_TABLE];
+  static ref TABLE_PARAMETERS: Parameters = vec![
+    (
+      cstr!("Rows"),
+      cstr!("Sequence of shards to build the rows."),
+      &SEQ_OF_SHARDS_TYPES[..]
+    )
+      .into(),
+    (
+      cstr!("Columns"),
+      cstr!("Configuration of the columns."),
+      &SEQ_OF_ANY_TABLE_TYPES[..]
+    )
+      .into(),
+    (
+      cstr!("Striped"),
+      cstr!("Whether to alternate a subtle background color to every other row."),
+      BOOL_VAR_OR_NONE_SLICE,
+    )
+      .into(),
+    (
+      cstr!("Resizable"),
+      cstr!("Whether columns can be resized within their specified range."),
+      BOOL_VAR_OR_NONE_SLICE,
+    )
+      .into(),
+    (
+      cstr!("RowIndex"),
+      cstr!("Variable to hold the row index, to be used within Rows."),
```

Comment:
seems added complexity, can it be avoided somehow?

Reply:
It's optional. But there is no other way to tell a row calback in which index it is when it is called.